### ISSUE TITLE
Rustfmt inserts a made-up `impl Trait` definition into types

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rustfmt#5086]
As of current master, rustfmt applies the following incorrect diff to this valid Rust code:

```diff
  #[cfg(any())]
- type Type: Bound;
+ type Type: Bound = impl Bound;
```

To reproduce: `cargo run --bin rustfmt repro.rs`

**Comment by calebcartwright** [comment]
Thanks for the report. Having a major case of déjà vu with this one

**Comment by ytmimi** [comment]
## Summary
Was doing a little digging into this.

TLDR; I think the issue is caused by using the ``bounds`` for both the ``generic_bounds_opt`` and as the ``OpaqueType`` used for the ``rhs`` when calling ``items::rewrite_ty``.

## Bug Details

When calling ``items::rewrite_type_alias`` we hit the first match_arm:

https://github.com/rust-lang/rustfmt/blob/eee8f0419dca910187350ac38fe9694b8b7919b9/src/items.rs#L1523-L1527

After we add the generic bounds to the output result (code responsible for that added bellow) we end up with the string ``type Type: Bound``

https://github.com/rust-lang/rustfmt/blob/eee8f0419dca910187350ac38fe9694b8b7919b9/src/items.rs#L1573-L1580

Because we have a ``rhs`` we end up adding a ``=`` to the end of the result string (code bellow) so now we have
``type Type: Bound =``
https://github.com/rust-lang/rustfmt/blob/eee8f0419dca910187350ac38fe9694b8b7919b9/src/items.rs#L1615-L1636

And lastly because of how ``OpaqueType::rewrite`` is implemented (code bellow) we end up adding the ``impl Bound``. before ultimately returning ``type Type: Bound = impl Bound;`` from ``items::rewrite_ty``.

https://github.com/rust-lang/rustfmt/blob/eee8f0419dca910187350ac38fe9694b8b7919b9/src/items.rs#L1874-L1881

## Proposed Solution

After going through the code it seems to me that passing ``None`` as the ``rhs`` argument of ``items::rewrite_ty``, would fix the issue, and locally it seems to have done the trick. I'm going to open a PR, but please let me know if I'm overlooking anything.
